docs/modules/ROOT/pages/BasicConcept.adoc

```diff
@@ -36,56 +53,180 @@ filter4 = "Power Nice" -- nick filter maches noone
+
+==== Step 1: Finding the Component
+
+The first step in your code is to find the Constructor on the network and save a *reference* to it in a variable. By saving a reference, we can then refer to its respective functions and properties very easily.
+
+This is accomplished by using the `component.proxy` function. We can use the function by itself and feed it the GUID of the Constructor in order to save a reference to it:
+
+`constructor = component.proxy('0123456789...')`
+
+But we used the Network Manager to name our Constructor "WireFactory Constructor1", and we'd like to use that nick instead of a long string of random numbers when we search for our component. Fortunately, we can let `component.findComponent` find the GUID for us using our nicks.
+
+`constructor = component.proxy(component.findComponent(
+    "WireFactory Constructor1")[1])`
+
+Is it pretty? No. Is it functional? Yes. 
+
+The function returns an array containing GUIDs of every component in the network whose nick matched the query string. Our query string should only return our one Constructor, so now we just need to reference the first element in the array. We add `[1]` at the end of our `findComponent` function call in order to reference the first item it returns. 
+
+_Note: in Lua, arrays start at one, which is unlike most programming languages._
+
+==== Step 2: Using the 'getRecipe' Function
+
+Now that we have a reference to our Constructor saved in the variable `constructor`, we can reference its functions and properties with very little effort.
+
+A look at the description for the https://docs.ficsit.app/ficsit-networks/latest/Reflection.html#Class-Manufacturer[Manufacturer component] informs us that it is actually the base class for _all_ machines that use a recipe, including our Constructor. We don't have to scroll much farther to see that `getRecipe` is the function in that class that we need to achieve our goal!
+
+Looking at the description of `getRecipe`, we see the following:
+
+"Get Recipe `getRecipe` (Class(Recipe) Recipe `recipe` _out_)"
+
+This might look like gibberish at first, but here's a look at each of the pieces individually:
+
+- *Get Recipe* - More readable, "localized name" of the function
+- `getRecipe` - Name of the function within the Reflection System
+- *()* - Parentheses indicate a list of the inputs and outputs of the function (known as "arguments") separated by commas
+- *Class(Recipe)* - Type of the argument. In this case, the type of the output argument is the `Recipe` class
+- *Recipe* - Localized name of the argument
+- `recipe` - Name of the argument in the Reflection System
+- _out_ - Indicates that the argument is an output
+
+Thanks to this information, we now know that the function `getRecipe` has an output argument called `recipe`, which is exactly what we need! 
+
+We can call the function by using something called "dot notation":
+
+`constructor.getRecipe()`
```

Comment:
This is not fully true... yes we can calle the function like that, but functions are not bound to any object...
there is no difference between a "static function" or a "member function/method"...
instead by the Lua convention for methods, we expect the first passed argument to be the instance itself...
because `constructor.getRecipe` essentially returns the function alone, after that it will be called when you do the `()`.
So... after Lua convention it should be `constructor.getRecipe(constructor)` but that is ugly and annoying, so Lua provides a pure syntactical sugar `constructor:getRecipe()` that does the same.
So please use `:` instead.